lib/vanagon/project.rb

```diff
@@ -476,5 +477,36 @@ def generate_package
     def generate_packaging_artifacts(workdir)
       @platform.generate_packaging_artifacts(workdir, @name, binding)
     end
+
+    # Generate a json file which lists all of the dependant components of the project
+    def generate_dependencies_info
+      project_components = @components.map do |component|
+        components_hash = component.options
+        [:version, :url].each do |key|
+          if val = component.send(key)
+            components_hash[key] = val
+          end
+        end
+        [
+          component.name,
+          components_hash,
+        ]
+      end.to_h
+    end
+
+    def generate_manifest_json
+      build_time = Time.now
+      manifest = {
+        "packaging_type" => {
+          "vanagon" => Gem.loaded_specs["vanagon"].version.to_s,
+        },
+        "git_sha" => @version,
```

Comment:
1. use the accessor (`#version`, not `@version`)
2. is `#version` always going to be a Git SHA?
3. what will happen if `#version` is neither a String nor a Git SHA?